lib/spack/spack/fetch_strategy.py

```diff
@@ -421,6 +430,79 @@ def fetch(self):
         tty.msg('Using cached archive: %s' % path)
 
 
+class PyPiFetchStrategy(URLFetchStrategy):
+    """Fetches packages directly from PyPi using XML-RPC to query the API."""
+
+    enabled = True
+    required_attributes = ['pypi']
+
+    def __init__(self, pypi=None, digest=None, **kwargs):
+        # If pypi or digest are provided in the kwargs, then prefer
+        # those values.
+        self.pypi = kwargs.get('pypi', None)
+        if not self.pypi:
+            self.pypi = pypi
+
+        self.version = kwargs.get('version', None)
+
+        self.digest = kwargs.get('md5', None)
+        if not self.digest:
+            self.digest = digest
+
+        self.expand_archive = kwargs.get('expand', True)
```

Comment:
Maybe we can fix `URLFetchStrategy` too.  But even if we don't, no need to propagate this stuff.